api/agent/agent.go

```diff
@@ -1045,7 +822,10 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 			}
 
 		case <-ctx.Done():
-			call.slots.queueSlot(&hotSlot{done: make(chan struct{}), fatalErr: ctx.Err()})
+			// XXX(reed): this seems like a bad idea? why are we even handing out a
+			// bad slot? shouldn't we make the client wait for a valid one and maybe
+			// timeout? not in this PR, NOT TONIGHT!
+			call.slots.queueSlot(&hotSlot{done: make(chan struct{}), fatalErr: models.ErrContainerExitedEarly})
```

Comment:
@skinowski we should discuss this behavior maybe somewhere else, this PR is a stinky enough turd without getting into the weeds on this behavior here, just highlighting...